src/3d/qgsframegraph.cpp

```diff
@@ -796,6 +802,25 @@ void QgsFrameGraph::setupDirectionalLight( const QgsDirectionalLightSettings &li
   mPostprocessingEntity->setupDirectionalLight( lightPosition, lightDirection );
 }
 
+QString QgsFrameGraph::dumpFrameGraph() const
+{
+  QObject *top = mRenderSurfaceSelector;
+  while ( top->parent() && dynamic_cast<Qt3DRender::QFrameGraphNode *>( top->parent() ) )
+    top = top->parent();
+
+  QgsFgUtils::FgDumpContext context;
+  context.lowestId = mMainCamera->id().id();
+  QStringList strList = QgsFgUtils::dumpFrameGraph( dynamic_cast<Qt3DRender::QFrameGraphNode *>( top ), context );
+
+  return qPrintable( strList.join( "\n" ) ) + QString( "\n" );
```

Comment:
Why the `qPrintable` on those two return values?

Reply:
ok